Migrate state mutations (block/uncle rewards, dao) to params/mutations

iquidus opened the pull request:
Introduces a new "mutations" module (found in params/mutations/*), intended for state mutations that take place outside of standard transactions, e.g block/uncle rewards, dao.

This logic has traditionally been located within the consensus/ethash and consensus/misc modules. Given these state mutations are likely to vary between chains (And are not technically part of ethash) it makes more sense to keep them in params.

No changes have been made to chain configs. 

reference #20

meowsbits reviewed (commented):
I'm not sure the `--mainnet` flag commit belongs in this branch, or? 

https://github.com/etclabscore/core-geth/pull/335/commits/4ed1399fd4f577be676fb06d2ddffe2e0032cddf

holiman committed:
cmd/geth, eth/downloader: remove copydb command (#22501)

* cmd/geth: remove copydb command

* eth/downloader: remove fakepeer

 Conflicts:
	cmd/geth/chaincmd.go

meowsbits reviewed (approved):
Looks fine to me. Per a team chat yesterday, there may be some unit tests for rewards in the works. Up to you @iquidus if you want to include those in the PR or a later one.

meowsbits committed:
Revert "tests: temporarily disable anything except test config file writer"

This reverts commit 54326b90f094395931063330b5cdf67f04716b70.

meowsbits committed:
Revert "Revert "tests: temporarily disable anything except test config file writer""

This reverts commit f4bf39cdcb87535d7df4e97a6c0e9a46f0364ab6.

meowsbits committed:
tests: squashme

Date: 2021-03-24 15:06:51-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: improve error handling

Date: 2021-03-24 15:13:05-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: simplify generative test runner

Date: 2021-03-24 15:17:16-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Makefile,tests: use log instead of t.Log, -p = 1

Date: 2021-03-24 15:22:32-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Makefile: -p = 1 is already set by default

Date: 2021-03-24 15:23:48-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Makefile,tests: wip

Date: 2021-03-24 15:56:19-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm,tests: still tinkering with exploring why write ok but not read ok

Date: 2021-03-24 16:33:29-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: only generate Phoenix state tests

Date: 2021-03-24 16:34:03-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: fix metadata filling

Date: 2021-03-24 16:46:50-05:00
Signed-off-by: meows <b5c6@protonmail.com>

holiman committed:
eth/tracers, core: use scopecontext in tracers, provide statedb in capturestart (#22333)

Fixes the CaptureStart api to include the EVM, thus being able to set the statedb early on. This pr also exposes the struct we used internally in the interpreter to encapsulate the contract, mem, stack, rstack, so we pass it as a single struct to the tracer, and removes the error returns on the capture methods.

holiman committed:
core: fix condition on header verification

fjl committed:
cmd/devp2p: fix comparison of TXT record value (#22572)

* cmd/devp2p: fix comparison of TXT record value

The AWS API returns quoted DNS strings, so we must encode the new value
before comparing it against the existing record content.

* cmd/devp2p: add test

* cmd/devp2p: fix typo and rename val -> newValue

karalabe committed:
Merge pull request #22573 from holiman/checkfreq

core: fix condition on header verification

meowsbits reviewed (changes requested):
Eventually I'd suggest the commit message https://github.com/etclabscore/core-geth/pull/335/commits/41fa2e43a950a0df6b5a34ada81db8044d3e65cf include more information.

meowsbits committed:
tests: [debug]: move chainids -> 1 for phoenix,magneto

Date: 2021-03-25 08:31:37-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: generate all tests

Date: 2021-03-25 08:32:59-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: add 'indexes' json key

Date: 2021-03-25 09:30:47-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: reset etc chain id -> 61

Date: 2021-03-25 09:31:03-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: comment

Date: 2021-03-25 09:31:10-05:00
Signed-off-by: meows <b5c6@protonmail.com>

MariusVanDerWijden committed:
eth: dump rpc gas cap and tx fee cap (#22574)

meowsbits committed:
Makefile,tests: run test from file right after generating

Date: 2021-03-25 13:33:51-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Makefile: workaround test timeout flag (unavailable in ci.go)

Date: 2021-03-25 13:34:58-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: skip long tests for now

Date: 2021-03-25 13:35:54-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: whitelist stZero

Date: 2021-03-25 13:52:39-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: debug print orig/gend test json

Date: 2021-03-25 14:03:15-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: use t.log instead of log.print

Date: 2021-03-25 14:03:57-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: write files, not log

Date: 2021-03-25 14:05:11-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: replace 2929 logic, but expect no effect

Date: 2021-03-25 14:18:01-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Revert "tests: replace 2929 logic, but expect no effect"

This reverts commit bd4fd85027182d67c6e5b35ca82c29e3b7d6619a.

meowsbits committed:
tests: debugging, dont use t.Run

Date: 2021-03-25 14:31:46-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: dont run in parallel

Date: 2021-03-25 14:32:44-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: dont skip anything; might work ok without parallelism

Date: 2021-03-25 14:37:18-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: extract to func

Date: 2021-03-25 14:42:30-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: add extra condition for filledPostStates

getting empty hashes for expected post state roots
somehow...

Date: 2021-03-25 14:46:38-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: copy ref->target post slice

Date: 2021-03-25 14:52:35-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: debugginggggg

Date: 2021-03-25 16:51:09-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: dont rerun tests right away

Date: 2021-03-25 16:55:20-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: fixup

Date: 2021-03-25 16:55:43-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: fix crossReference post state

Date: 2021-03-25 17:16:00-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: paranoid about shadows

Date: 2021-03-25 17:18:25-05:00
Signed-off-by: meows <b5c6@protonmail.com>

karalabe committed:
eth/protocols, metrics, p2p: add handler performance metrics

meowsbits committed:
tests: rewrite test generator

Date: 2021-03-26 07:23:53-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Makefile: use new test generator test fn

Date: 2021-03-26 07:25:02-05:00
Signed-off-by: meows <b5c6@protonmail.com>

karalabe committed:
Merge pull request #22581 from karalabe/network-latency-monitoring

eth/protocols, metrics, p2p: add handler performance metrics

meowsbits committed:
tests: fix? oob err stPostState slice

Date: 2021-03-26 07:30:20-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: gen callback on write

Date: 2021-03-26 07:34:13-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: refactor for readibility

Date: 2021-03-26 07:54:18-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: skip genstate if no env vars set

Date: 2021-03-26 07:58:56-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: allow parallel

Date: 2021-03-26 07:59:43-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: loop test dirs

Date: 2021-03-26 08:02:59-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: parallel called multiple times

Date: 2021-03-26 08:05:32-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: disable parallel again

Date: 2021-03-26 08:07:36-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: revert test run layout

Date: 2021-03-26 08:26:17-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: testMatcher field to disable parallelism

Date: 2021-03-26 08:32:49-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Makefile,tests: dont use parity chainspecs

Date: 2021-03-26 08:44:02-05:00
Signed-off-by: meows <b5c6@protonmail.com>

karalabe committed:
eth/protocols, metrics: use resetting histograms for rare packets

karalabe committed:
Merge pull request #22586 from karalabe/resettting-histogram

eth/protocols, metrics: use resetting histograms for rare packets

meowsbits committed:
tests: single file test gener

Date: 2021-03-26 10:37:02-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: wip/save

Date: 2021-03-26 10:43:01-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: wip/save

Date: 2021-03-26 11:16:41-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: wip/save

Date: 2021-03-26 11:19:11-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Makefile,tests: wip/save

Date: 2021-03-26 11:23:19-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: write generated test files to new dir

I was seeing wrongly-encoded JSON data in the
overwritten test files, so this should avoid
reads and writes colliding.

Date: 2021-03-26 11:42:02-05:00
Signed-off-by: meows <b5c6@protonmail.com>

zouguangxian committed:
eth: fix corner case in sync head determination (#21695)

This avoids synchronisation failures when the local header is ahead of
the local full block.

fjl committed:
cmd/geth, consensus/ethash: add support for --miner.notify.full flag (#22558)

The PR implements the --miner.notify.full flag that enables full pending block
notifications. When this flag is used, the block notifications sent to mining
endpoints contain the complete block header JSON instead of a work package
array.

Co-authored-by: AlexSSD7 <alexandersadovskyi7@protonmail.com>
Co-authored-by: Martin Holst Swende <martin@swende.se>

meowsbits committed:
tests: add on-skip callback to remove noop files; rename gend files

Date: 2021-03-26 13:30:41-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: REVERTME skip empty generated files

Date: 2021-03-26 13:32:05-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Revert "tests: REVERTME skip empty generated files"

This reverts commit 617446cd179a1c0a9b512cbdda3e35a0bbd2960a.

meowsbits committed:
tests: generated State tests

Date: 2021-03-26 13:53:18-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: refactor generated temporary file logic

Date: 2021-03-26 14:01:18-05:00
Signed-off-by: meows <b5c6@protonmail.com>

karalabe committed:
metrics/influxdb: don't push empty histograms, no measurement != 0

karalabe committed:
Merge pull request #22590 from karalabe/nil-histogram

metrics/influxdb: don't push empty histograms, no measurement != 0

karalabe committed:
eth/protocols/snap: add peer id and req id to the timeout logs

karalabe committed:
Merge pull request #22591 from karalabe/snap-detailed-timeout-logs

eth/protocols/snap: add peer id and req id to the timeout logs

fjl committed:
cmd/devp2p: update to newer cloudflare API client (#22588)

This upgrades the cloudflare client dependency to v0.14.0. The new
version changes the API because all methods now require a context
parameter. This change also reduces the log level of the 'Skipping...'
message to debug, following a similar change in the AWS deployer.

meowsbits committed:
params/types/tests: test convert back-and-forth parity:coregeth

Date: 2021-03-26 18:44:37-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/coregeth.json.d,params/parity.json.d,params/types/tests,tests: generate parity state test configs

Date: 2021-03-27 09:54:02-05:00
Signed-off-by: meows <b5c6@protonmail.com>

rjl493456442 committed:
core/state/pruner: move the compaction out of the pruning procedure (#22579)

The main idea behind it is: the range compaction is very expensive
which can take a few hours to finish. During this long procedure,
a lot of exceptions can occur, e.g.
- Geth is killed manually
- Geth is killed because of machine crash
- etc

In order to minimize the effect of the exceptions, the compaction
is moved out of the pruning. So that even the compaction is not
finished, the pruning is regarded as done.

holiman committed:
eth/protocols/snap: try to prevent requests timing out

renaynay committed:
core: add BlockGen.GetBalance method  (#22589)

MariusVanDerWijden committed:
cmd/puppeth: specify working directory for nodejs 15 (#22549)

meowsbits committed:
tests: (lint) remove unused variables

Date: 2021-03-29 15:13:34-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/types/tests: fix comparator logic (thanks ineffassign!)

Date: 2021-03-29 15:16:17-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: (lint/govet): remove json tag for nonexported struct

Date: 2021-03-29 15:16:59-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Merge branch 'master' into merge/foundation-release/1.10.1-resolved

Conflicts:
      cmd/geth/consolecmd_test.go
      ethclient/ethclient_test.go
      go.mod
      go.sum

meowsbits committed:
cmd/geth: fixup network identites test to use lates yolov3 params

Date: 2021-03-29 15:23:43-05:00
Signed-off-by: meows <b5c6@protonmail.com>

karalabe committed:
Merge pull request #22596 from holiman/snap_timelimits

eth/protocols/snap: try to prevent requests timing out

holiman committed:
cmd/geth: add db dumptrie command (#22563)

Adds the command "geth db dumptrie <root> <seek> <max>", to better help investigate the trie data

meowsbits committed:
build: bump test timeout to 30m

Tests timed out at https://travis-ci.org/github/etclabscore/core-geth/jobs/764945270

Date: 2021-03-30 07:00:37-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: skip generative tests

Date: 2021-03-30 08:00:05-05:00
Signed-off-by: meows <b5c6@protonmail.com>

nebojsa94 committed:
core/vm: fix Byzantium address list (#22603)

meowsbits committed:
Makefile: remove parity config testing

Date: 2021-03-30 08:46:44-05:00
Signed-off-by: meows <b5c6@protonmail.com>

MariusVanDerWijden committed:
ethstats: avoid creating subscriptions on background goroutine (#22587)

This fixes an issue where the ethstats service could crash if geth was
started and then immediately stopped due to an internal error. The
cause of the crash was a nil subscription being returned by the backend,
because the background goroutine creating them was scheduled after
the backend had already shut down.

Moving the creation of subscriptions into the Start method, which runs
synchronously during startup of the node, means the returned subscriptions
can never be 'nil'.

Co-authored-by: Felix Lange <fjl@twurst.com>
Co-authored-by: Martin Holst Swende <martin@swende.se>

meowsbits committed:
Merge branch 'master' into merge/foundation-release/1.10.1-resolved

Conflicts:
      core/vm/evmc.go
      go.mod
      go.sum

meowsbits committed:
core/vm,tests: skip Berlin tests for EVMC (v7 does not support Berlin)

Date: 2021-03-30 10:10:19-05:00
Signed-off-by: meows <b5c6@protonmail.com>

karalabe committed:
core/state/snapshot, ethdb: track deletions more accurately (#22582)

* core/state/snapshot, ethdb: track deletions more accurately

* core/state/snapshot: don't reset the iterator, leveldb's screwy

* ethdb: don't mess with the insert batches for now

meowsbits committed:
tests: regex syntax error

Date: 2021-03-30 11:21:48-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Makefile: bump timeouts on cconfig equivalence tests

Date: 2021-03-30 12:39:46-05:00
Signed-off-by: meows <b5c6@protonmail.com>

fjl committed:
rpc: tighter shutdown synchronization in client subscription (#22597)

This fixes a rare issue where the client subscription forwarding loop
would attempt send on the subscription's channel after Unsubscribe has
returned, leading to a panic if the subscription channel was already
closed by the user. Example:

    sub, _ := client.Subscribe(..., channel, ...)
    sub.Unsubscribe()
    close(channel)

The race occurred because Unsubscribe called quitWithServer to tell the
forwarding loop to stop sending on sub.channel, but did not wait for the
loop to actually come down. This is fixed by adding an additional channel
to track the shutdown, on which Unsubscribe now waits.

Fixes #22322

meowsbits committed:
.travis.yml: travis build timed out

https://travis-ci.org/github/etclabscore/core-geth/jobs/765123692

Date: 2021-03-30 16:46:14-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
.travis.yml,build: travis still timing out

Date: 2021-03-30 18:34:03-05:00
Signed-off-by: meows <b5c6@protonmail.com>

karalabe committed:
all: fix miner hashRate -> hashrate on API calls

meowsbits committed:
build: syntax error

Date: 2021-03-31 06:29:42-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: use ioutil temp file instead of handrolled junk

Date: 2021-03-31 06:45:50-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: commentary

Date: 2021-03-31 06:57:24-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: commentary for test generation funcs

Date: 2021-03-31 07:08:19-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: remove core-geth-specific metadata fields from state test schema

Date: 2021-03-31 07:08:40-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: remove antiquated code (first generation tests-generation code)

Date: 2021-03-31 07:15:05-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Makefile,tests: init sketch for writing chain configs with generated tests

Stopping here because I'm realizing that
its going to be more complicated than just
writing one genesis+chain config for each
Fork.

Each StateTest defines its own genesis env,
so that each test can have its own genesis.

We have to write ONLY chain configs. Not genesis.

Date: 2021-03-31 07:32:33-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: write chain configs for all test forks

Use only chain config values (not genesis).

Date: 2021-03-31 07:39:58-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: reset testdata to foundation-1.10.1

Date: 2021-03-31 07:41:20-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: difficulty test gen uses core-geth chain config type

Date: 2021-03-31 07:56:14-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: syntax error

Date: 2021-03-31 07:57:03-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: add return var names for readability

Date: 2021-03-31 08:28:27-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: hack in core-geth config file pathing (akin parity config file specs)

Date: 2021-03-31 08:49:56-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: bump testdata to generated state tests commit

Date: 2021-03-31 08:51:03-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Makefile: use core-geth chainspecs in difficulty test gen

Date: 2021-03-31 08:53:41-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: fix hyphenation typo in core-geth config dir path name

Date: 2021-03-31 08:55:02-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: use core-geth chainspecs env instead of go core-geth strut env

Date: 2021-03-31 08:55:58-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/coregeth.json.d,params/parity.json.d,tests: generate core-geth chainspec'd difficulty tests

Date: 2021-03-31 08:58:09-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Makefile: add command to test all tests with core-geth chainspecs

Date: 2021-03-31 08:58:25-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Makefile,params/coregeth.json.d,tests: reset to generate difficulty again

Running difficulty tests yields missing gaslimit
for genesiss type, which means the file only contains
the chain config and not the genesis values.

I assume this is because there were pre-existing files
which were not overwritten with genesis information.

Date: 2021-03-31 09:11:47-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/coregeth.json.d,tests: generate difficulty tests

Date: 2021-03-31 09:12:16-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: write ropsten genesis

Date: 2021-03-31 09:13:56-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/coregeth.json.d: remove coregeth configs

Date: 2021-03-31 09:14:08-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: reset testdata, removing difficulty

Date: 2021-03-31 09:14:28-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/coregeth.json.d,tests: generate difficulty tests

Date: 2021-03-31 09:14:53-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: :gun:

Date: 2021-03-31 09:21:54-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/coregeth.json.d,tests: reset generated difficulty tests

Date: 2021-03-31 09:22:31-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
:hammer:

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
:hammer:

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
:hammer:

Signed-off-by: meows <b5c6@protonmail.com>

karalabe committed:
Merge pull request #22604 from karalabe/hashrate-capitalization-fix

all: fix miner hashRate -> hashrate on API calls

whilei committed:
params,Makefile,tests: OK: generate difficulty tests

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/types/coregeth: add EIP1234 respective conditions for EIP649

This fixes and issue where setting 649 after 1234
would overwrite with wrong block rewards (3 intead
of 2).

Date: 2021-03-31 17:58:48-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
:hammer:

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/coregeth.json.d: regenerate state core-geth configs

Date: 2021-03-31 18:54:11-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/types/multigeth: remove multigeth special handling of EIP1283/Disable interaction

This fixes a failing test TestGeneratedConfigsEq
when run with multigeth chainspecs.

The SSTORE implemented gas cost will work
OK without this special handling.

Date: 2021-03-31 19:15:38-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: add forkname to error for debugging

Date: 2021-03-31 19:15:50-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: commentary for new tests

Date: 2021-04-01 03:51:44-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: commentary and nice-ificiation

Date: 2021-04-01 04:13:50-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: (lint) fixup linting errors, eg. removing unused things

Date: 2021-04-01 04:17:59-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Makefile: remove deprecated makefile command for parity

Date: 2021-04-01 04:32:24-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params,params/types/multigeth,tests: fixup tests w/r/t EIP1283/Disable transations

Notably these tests need to work for the matrix
of all configuration schemas (eg. Multigeth) which
is not always super pretty.

The special case for EIP1283 is handled specific to the
troublesome/indicative test for now; it could also
be handled at the Configurator method level
as a general case, but that's going too deep for me right
now.

Date: 2021-04-01 05:50:23-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: reset to ethereum/go-ethereum@1.10.1

Date: 2021-04-01 06:43:25-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Makefile: fix typo

Date: 2021-04-01 06:44:18-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/coregeth.json.d,tests: generate difficulty tests and test configs

Date: 2021-04-01 06:46:31-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: generate state tests and configs

Date: 2021-04-01 08:05:14-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/types/ctypes: test should actually test asserted EIP1234 match

Date: 2021-04-01 08:25:26-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: maybe unmarshal tests preserving name

Date: 2021-04-01 17:11:08-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: reset tests to ethereum/go-ethereum@1.10.1

Date: 2021-04-01 17:11:36-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: bump to generated difficulty tests

Date: 2021-04-01 17:12:37-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Revert "tests: bump to generated difficulty tests"

This reverts commit 977f5c3549bef45569431180c876df77890f7140.

meowsbits committed:
Revert "Revert "tests: bump to generated difficulty tests""

This reverts commit d74187e27baa434df25dc3360b3f5bf2836ed4ab.

meowsbits committed:
Revert "tests: maybe unmarshal tests preserving name"

This reverts commit f8ee352a480a0ccc7daee8a8a9b5681037770d96.

meowsbits committed:
tests: try different way of naming

Date: 2021-04-01 17:22:35-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: bump to generated state tests

Date: 2021-04-01 17:33:45-05:00
Signed-off-by: meows <b5c6@protonmail.com>

iquidus committed:
mutations: add more tests

iquidus commented on params/mutations/rewards_test.go:
```diff
@@ -192,3 +214,335 @@ func TestGetBlockWinnerRewardForUnclesByEra(t *testing.T) {
+		db.Close()
+	}
+}
+*/
```
This test (from the old code), still needs resolving. It runs (after refactor), but fails massively, further study required.

iquidus replied:
resolved.

iquidus commented on params/mutations/rewards_test.go:
```diff
@@ -192,3 +214,335 @@ func TestGetBlockWinnerRewardForUnclesByEra(t *testing.T) {
+	if uncleReward.Cmp(big.NewInt(64000000000000000)) != 0 {
+		t.Error("Should return uncleReward 64000000000000000", "reward", uncleReward)
+	}
+}
```
duplicate tests to a degree, however I figure more never hurts (ported over from core-pool).

meowsbits committed:
params/types/goethereum: go-ethereum config Stringer logs Berlin block

Date: 2021-04-05 05:33:55-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: add configuration files to test data

Date: 2021-04-05 06:18:29-05:00
Signed-off-by: meows <b5c6@protonmail.com>

fxfactorial committed:
core/state/snapshot: fix data race in diff layer (#22540)

piersy committed:
internal/ethapi: fix eth_chainId method (#22243)

This removes the duplicated definition of eth_chainID
in package eth and updates the definition in internal/ethapi
to treat chain ID as a bigint.

Co-authored-by: Felix Lange <fjl@twurst.com>

AmitBRD committed:
graphql: add support for tx types and tx access lists (#22491)

This adds support for EIP-2718 access list transactions in the GraphQL API.

Co-authored-by: Amit Shah <amitshah0t7@gmail.com>
Co-authored-by: Felix Lange <fjl@twurst.com>

s1na committed:
internal/debug: add JSON log format and rename logging flags (#22341)

This change adds support for logging JSON records when the --log.json flag is
given. The --debug and --backtrace flags are deprecated and replaced by
--log.debug and --log.backtrace.

While changing this, it was noticed that the --memprofilerate and
--blockprofilerate were ineffective (they were always overridden even if
--pprof.memprofilerate was not set). This is also fixed.

Co-authored-by: Felix Lange <fjl@twurst.com>

petesimard committed:
cmd/utils: move cache sanity check to SetEthConfig (#22510)

Move the cache sanity check to the SetEthConfig function
to allow the config file to load.

Evolution404 committed:
consensus/ethash: replace a magic number with it's constant (#22618)

zsfelfoldi committed:
les: move client pool to les/vflux/server (#22495)

* les: move client pool to les/vflux/server

* les/vflux/server: un-expose NodeBalance, remove unused fn, fix bugs

* tests/fuzzers/vflux: add ClientPool fuzzer

* les/vflux/server: fixed balance tests

* les: rebase fix

* les/vflux/server: fixed more bugs

* les/vflux/server: unexported NodeStateMachine fields and flags

* les/vflux/server: unexport all internal components and functions

* les/vflux/server: fixed priorityPool test

* les/vflux/server: polish balance

* les/vflux/server: fixed mutex locking error

* les/vflux/server: priorityPool bug fixed

* common/prque: make Prque wrap-around priority handling optional

* les/vflux/server: rename funcs, small optimizations

* les/vflux/server: fixed timeUntil

* les/vflux/server: separated balance.posValue and negValue

* les/vflux/server: polish setup

* les/vflux/server: enforce capacity curve monotonicity

* les/vflux/server: simplified requestCapacity

* les/vflux/server: requestCapacity with target range, no iterations in SetCapacity

* les/vflux/server: minor changes

* les/vflux/server: moved default factors to balanceTracker

* les/vflux/server: set inactiveFlag in priorityPool

* les/vflux/server: moved related metrics to vfs package

* les/vflux/client: make priorityPool temp state logic cleaner

* les/vflux/server: changed log.Crit to log.Error

* add vflux fuzzer to oss-fuzz

Co-authored-by: rjl493456442 <garyrong0905@gmail.com>

rjl493456442 committed:
eth, les: fix tracers (#22473)

* eth, les: fix tracer

* eth: isolate live trie database in tracer

* eth: fix nil

* eth: fix

* eth, les: add checkLive param

* eth/tracer: fix

ziogaschr committed:
Merge branch 'master' into merge/foundation-release/1.10.1-resolved

# Conflicts:
#	core/state_transition.go
#	core/vm/logger.go
#	eth/tracers/api.go
#	eth/tracers/api_parity.go
#	eth/tracers/internal/tracers/assets.go
#	eth/tracers/internal/tracers/prestate_tracer.js
#	eth/tracers/tracer.go

meowsbits committed:
cmd/echainspec: bump key -> yolov3 for chainspec map

Date: 2021-04-07 07:34:47-05:00
Signed-off-by: meows <b5c6@protonmail.com>

MariusVanDerWijden committed:
core, eth, internal/ethapi: create access list RPC API (#22550)

* core/vm: implement AccessListTracer

* eth: implement debug.createAccessList

* core/vm: fixed nil panics in accessListTracer

* eth: better error messages for createAccessList

* eth: some fixes on CreateAccessList

* eth: allow for provided accesslists

* eth: pass accesslist by value

* eth: remove created acocunt from accesslist

* core/vm: simplify access list tracer

* core/vm: unexport accessListTracer

* eth: return best guess if al iteration times out

* eth: return best guess if al iteration times out

* core: docstring, unexport methods

* eth: typo

* internal/ethapi: move createAccessList to eth package

* internal/ethapi: remove reexec from createAccessList

* internal/ethapi: break if al is equal to last run, not if gas is equal

* internal/web3ext: fixed arguments

* core/types: fixed equality check for accesslist

* core/types: no hardcoded vals

* core, internal: simplify access list generation, make it precise

* core/vm: fix typo

Co-authored-by: Martin Holst Swende <martin@swende.se>
Co-authored-by: Péter Szilágyi <peterke@gmail.com>

karalabe committed:
eth: fix tracing state retrieval if requesting the non-dirty genesis

karalabe committed:
Merge pull request #22629 from karalabe/fix-chain-tracing-from-0

eth: fix tracing state retrieval if requesting the non-dirty genesis

meowsbits committed:
build: use env-aware bash shebangs for evmc-build scripts

Date: 2021-04-07 11:16:10-05:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
ethclient: add `debug_traceCallMany` in discovery tests

ziogaschr committed:
ethclient: remove duplicate rpc method name in tests

karalabe committed:
params: update CHTs for v1.10.2

karalabe committed:
Merge pull request #22632 from karalabe/cht-1.10.2

params: update CHTs for v1.10.2

fjl committed:
params: release go-ethereum v1.10.2 stable

iquidus committed:
mutations: add TestAccumulateRewards

iquidus committed:
mutations: lint

ziogaschr committed:
ethclient, node: test for subscription methods in openrpc discovery

ziogaschr committed:
eth/tracers: make JS tracer tests use go-ethereum configuration so as they can merged upstream

meowsbits committed:
Merge remote-tracking branch 'etclabscore/merge/foundation-release/1.10.1-resolved-stateDiff' into merge/foundation-release/1.10.1-resolved

ziogaschr committed:
Merge pull request #355 from etclabscore/merge/foundation-release/1.10.1-resolved-stateDiff

Refactor/merge stateDiff from master on upstream release/1.10.1

meowsbits committed:
Merge branch 'foundation-1.10.2' into merge/foundation-release/1.10.2-resolved

 Conflicts:
	cmd/devp2p/internal/ethtest/suite.go
	cmd/devp2p/internal/ethtest/types.go
	cmd/geth/chaincmd.go
	cmd/geth/main.go
	cmd/puppeth/module_dashboard.go
	cmd/utils/flags.go
	consensus/ethash/algorithm.go
	consensus/ethash/algorithm_test.go
	consensus/ethash/ethash.go
	consensus/ethash/sealer_test.go
	core/rawdb/database.go
	core/rawdb/freezer.go
	core/state_transition.go
	core/tx_pool.go
	core/vm/contracts.go
	core/vm/eips.go
	core/vm/evm.go
	core/vm/instructions.go
	core/vm/logger.go
	core/vm/logger_json.go
	core/vm/runtime/runtime.go
	core/vm/runtime/runtime_test.go
	eth/api.go
	eth/backend.go
	eth/ethconfig/config.go
	eth/state_accessor.go
	eth/tracers/api.go
	eth/tracers/tracer.go
	go.mod
	go.sum
	internal/ethapi/api.go
	les/peer.go
	les/server.go
	params/config.go
	params/version.go

meowsbits committed:
go.mod,go.sum: go get -d ./...

Date: 2021-04-13 10:14:48-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
go.mod,go.sum: go mod tidy

Date: 2021-04-13 10:15:02-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/rawdb: fixup readonly vals for FreezerRemote

Date: 2021-04-13 10:17:44-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
internal/ethapi: fixup precompiles list construction for config/blocknum

Date: 2021-04-13 10:23:24-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: implement missing tracer interface method CapturePreEVM

Date: 2021-04-13 10:23:45-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: remove disused ActivePrecompiles method

We already can build a superset value of this
via PrecompiledContractsForConfig.

Date: 2021-04-13 10:24:57-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/puppeth: rm -rf cmd/puppeth

Upstream added a new dashboard html file.

We have purged puppeth.

Date: 2021-04-13 10:26:19-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/utils,node: fixup node database freezer remote readonly signature

Date: 2021-04-13 10:27:35-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
consensus/ethash: fixup ethash config construction sytax for test

Date: 2021-04-13 10:29:47-05:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
core/vm, eth/tracers: fix conflicts with Tracer interface

ziogaschr committed:
eth/tracers: fix tracer related conflicts

ziogaschr committed:
eth/ethconfig: fix failed/broken merge

ziogaschr committed:
cmd/echaindb: conform to NewLevelDBDatabase readonly mode

ziogaschr committed:
core/vm, eth/tracers: fix lint issues on tracers

ziogaschr committed:
core: fix remote freezer tests for readonly Database

meowsbits committed:
graphql: fixup core.Genesis -> genesisT.Genesis imports

Date: 2021-04-13 15:55:51-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
graphql: type ChainConfigurator has no field ChainID

Date: 2021-04-13 15:56:12-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: [PTAL] remove have/want from error string

I haven't figured out where this is getting
interpolated to the error value.

It seems the error is 'right', but is
not being formatted with the details.

Date: 2021-04-13 17:02:32-05:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
ethclient: fix TestRPCDiscover test

ziogaschr committed:
core/vm: apply CREATE|CREATE2 instruction gas usage logic

…broken because of the upstream merge

ziogaschr committed:
eth/tracers: continue using returnData in JS tracer CaptureState

It is needed by the callTracerParity for maintaining 100% OE compatibility

ziogaschr committed:
eth/tracers: call_tracer_parity to use the blockNumber passed in result

ziogaschr committed:
eth/tracers: use txTraceContext instead of taskExtraContext where possible

ziogaschr committed:
ethclient: remove whitespace at EOL

ziogaschr committed:
node: add openrpc schema mapper for subscription IDs

ziogaschr committed:
node: add missing title for rpcSubscriptionParamsNameD schema mapper

ziogaschr committed:
node: add openrpc schema mappers for debug, trace subscription methods

ziogaschr committed:
node: update openrpc schema mapper methods

ziogaschr reviewed (approved):
LGTM. I am waiting for sync to finish on classic on my side too.

ziogaschr committed:
node: whitelist skip pubsub to eth_ public api

I want to allow debug and trace API modules
through.

Taken from @meowsbits d7194b5

ziogaschr committed:
ethclient: generate docs for both `*_subscribe` as well Subscriptionable methods

ziogaschr committed:
docs/JSON-RPC-API/modules: run 'make docs-generate'

meowsbits committed:
core: simplify header chain nil check

Date: 2021-04-28 06:15:26-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
internal/debug: fix dropped deprecated (legacy) pprof flags

All depcrecated pprof flags were dropped, apparently
accidentally.

I noticed this because --debug was unavailable.

Not only were the flag declarations unregistered,
but logical support for pprof.port, addr, and cpuprofile
were missing; this adds support for them.

Date: 2021-04-28 06:33:15-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/geth: add classic,mordor,kotti flags for dbcmd and chaincmd

Date: 2021-04-28 06:52:00-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
build,cmd/echaindb: rm -rf and uses of echaindb command

This small program's functionality has been
replace with the new 'geth db' command.

Date: 2021-04-28 07:00:28-05:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
cmd/geth,cmd/utils,core,eth,eth/ethconfig: replace ecbp1100Flag and OverrideBerlinFlag usage with OverrideMagnetoFlag

ziogaschr committed:
ethclient: remove js console example from generated docs for subscrionable methods

ziogaschr committed:
docs/JSON-RPC-API/modules: run 'make docs-generate'

meowsbits committed:
Merge pull request #366 from etclabscore/merge/foundation-release/1.10.2-resolved-rmrf-cmd-echaindb

build,cmd/echaindb: rm -rf and uses of echaindb command

meowsbits committed:
ethclient: fixup test again canonical methods list

Adapts the canonical list of methods that
should be provided by the document to meet
what the document says.

Date: 2021-04-30 06:13:54-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: remove EIP2315 from Magneto overrides

This EIP was removed at the last minute from
the Berlin spec, and is -- pending merge --
slated for removal at the Magneto spec as well.

https://github.com/ethereumclassic/ECIPs/pull/419

Date: 2021-04-30 07:05:01-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: fixup comments (go fmt)

Date: 2021-04-30 07:11:38-05:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
cmd/geth,cmd/utils,eth/ethconfig: bring back the ECBP1100 flag

Based on https://github.com/etclabscore/core-geth/pull/367#discussion_r623792593

ziogaschr committed:
node: openrpc to return the `*filters.PublicFilterAPI` methods

ziogaschr committed:
ethclient: generated docs have `*filters.PublicFilterAPI` methods as well as better examples

ziogaschr committed:
docs/JSON-RPC-API/modules: run 'make docs-generate'

ziogaschr committed:
ethclient: bring up tests against canonical methods list, including `*filters.PublicFilterAPI` methods

ziogaschr committed:
Merge pull request #356 from etclabscore/merge/foundation-release/1.10.1-resolved-stateDiff-openrpc

Test for OpenRPC subscription methods on trace/debug modules

meowsbits committed:
Merge branch 'merge/foundation-release/1.10.1-resolved' into merge/foundation-release/1.10.2-resolved

meowsbits committed:
Merge pull request #357 from etclabscore/merge/foundation-release/1.10.2-resolved

Merge/foundation release/1.10.2 resolved

ziogaschr committed:
cmd/geth: add missing rawdb import

ziogaschr committed:
ethclient: remove `eth_hashrate` from openrpc test

meowsbits committed:
tests: comment explains absence of ETC_Magneto in gen'd difficulty tests

Date: 2021-05-06 06:25:57-05:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
eth: allow ecbp1100 activation block override

ziogaschr committed:
Merge pull request #367 from etclabscore/feat/magneto-flag

Add OverrideMagneto flag

ziogaschr committed:
cmd/utils, eth, eth/protocols/eth: fix allow selecting protocol versions after merge

meowsbits committed:
cmd/geth,internal/cmdtest: add KillTimeout field to TestCmd struct

TestGethStartupLogs was failing because of shutdown
timeout, via https://travis-ci.org/github/etclabscore/core-geth/jobs/770485427.

This allows a configurable process kill timeout window
and provides an unchanged default (5 seconds).

Note that the field isn't protected or really closely
integrated with the process runner at (ie via runGeth function),
so the kill field basically just gets set on the fly and
hopes that it gets set before the shutdown logic is called.

Date: 2021-05-11 05:56:29-05:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
Merge pull request #373 from etclabscore/merge/foundation-release/1.10.1-resolved-protocol-versions

Fix allowing of selecting protocol versions after merge of 1.10.x

meowsbits committed:
Merge pull request #375 from etclabscore/merge/foundation-release/1.10.1-resolved-testcmd-timeout

cmd/geth,internal/cmdtest: add KillTimeout field to TestCmd struct

meowsbits committed:
params: bump version from 1.11.23 -> 1.12.0-rc1

Date: 2021-05-11 15:34:55-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params: bump version from 1.12.0-rc1 to 1.12.0-rc1+

Date: 2021-05-11 15:40:50-05:00
Signed-off-by: meows <b5c6@protonmail.com>

iquidus committed:
Merge branch 'release/1.12-rc1' of github.com:etclabscore/core-geth into dev/rewards

meowsbits reviewed (changes requested):
Just giving a red review to remind that this PR has `master` as a base branch; I think it should want `rc1` as the base now.
